model/plugin_event.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+const (
+	PluginEventSendTypeReliable   = CLUSTER_SEND_RELIABLE
+	PluginEventSendTypeBestEffort = CLUSTER_SEND_BEST_EFFORT
+)
+
+// PluginEvent is used to allow intra-cluster plugin communication.
+type PluginEvent struct {
+	// Id is the unique identifier for the event.
+	Id string
+	// Data is the event payload.
+	Data []byte
```

Comment:
I have "confrontend feelings" here, I don't like all this back and forth with the `[]byte`/`string` difference between `PluginEvent` and `ClusterEvent`, but at the same time I think `[]byte` is the right type, but we don't want to break the backward compatibility. I just want to express my concern, but feel free to keep it or change it.

Reply:
I share the same feeling although I think we should move in the direction of making the the `ClusterEvent` field a `[]byte` instead. And even though it's in `model` it might be worth considering the modification. It's an enterprise only object which I am not sure has any outside use so I am thinking it could be changed without major side effects.

Reply:
This part will be handled in https://mattermost.atlassian.net/browse/MM-25116.